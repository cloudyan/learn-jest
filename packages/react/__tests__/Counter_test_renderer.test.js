import React from 'react';
import ReactDOM from 'react-dom';
import { act, create } from 'react-test-renderer';
import Counter from '../Counter';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it('can render and update a counter', () => {

  // 渲染组件
  let root;
  act(() => {
    root = create(<Counter />, container);
  });

  // 对根元素进行端言
  expect(root.toJSON()).toMatchSnapshot();

  expect(document.title).toBe('You clicked 0 times');
});
