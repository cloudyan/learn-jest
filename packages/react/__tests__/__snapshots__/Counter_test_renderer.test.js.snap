// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render and update a counter 1`] = `
<div>
  <p>
    You clicked 
    0
     times
  </p>
  <button
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
